File: test/java/DopTest/GsonTest/src/com/ffan/dop/BoolItem.java
package com.ffan.dop;

class BoolItem extends Item {
    /**
     * 值
     */
    private boolean value;

    /**
     * 构造函数
     */
    BoolItem(boolean value) {
        this.value = value;
        this.type = ItemType.BOOL_TYPE;
    }

    /**
     * 获取值int
     */
    public long getValueInt() {
        return this.value ? 1 : 0;
    }

    /**
     * 获取值bool
     */
    public boolean getValueBool() {
        return this.value;
    }

    /**
     * 获取值 string
     */
    public String getValueString() {
        return this.value ? "true" : "false";
    }

    /**
     * 获取值 double
     */
    public double getValueDouble() {
        return this.value ? 1.0 : 0.0;
    }
}

File: base/Protocol/BoolItem.php
<?php

namespace UiStd\Dop\Protocol;

use UiStd\Dop\Exception;

/**
 * Class BoolItem
 * @package UiStd\Dop\Protocol
 */
class BoolItem extends Item
{
    /**
     * @var int 类型
     */
    protected $type = ItemType::BOOL;

    /**
     * 设置默认值
     * @param string $value
     * @throws Exception
     */
    public function setDefault($value)
    {
        if (!is_string($value)) {
            throw new Exception('default value error');
        }
        $value = strtolower($value);
        $this->default = 'true' === $value ? 'true' : 'false';
    }
}
